src/main/java/in/koreatech/koin/domain/kakao/dto/KakaoSkillResponse.java

```diff
@@ -0,0 +1,85 @@
+package in.koreatech.koin.domain.kakao.dto;
+
+import static lombok.AccessLevel.PRIVATE;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import in.koreatech.koin.domain.kakao.exception.KakaoSkillFailedException;
+import lombok.NoArgsConstructor;
+
+/**
+ * <a href="https://kakaobusiness.gitbook.io/main/tool/chatbot/skill_guide/answer_json_format#skillresponse">
+ *     KAKAO 비즈니스 기술문서 - Bot-Skill 서버 연동 API
+ * </a>
+ */
+@NoArgsConstructor(access = PRIVATE)
+public class KakaoSkillResponse {
+
+    private static final int OUTPUTS_LIMIT = 3;
+    private static final int QUICK_REPLIES_LIMIT = 10;
+    private static final int MAX_QUICK_REPLIES_LABEL_SIZE = 8;
+
+    private static final String SIMPLE_TEXT = "simpleText";
+    public static final String QUICK_ACTION_MESSAGE = "message";
+
+    public static SkillResponseBuilder builder() {
+        return new SkillResponseBuilder();
+    }
+
+    public static class SkillResponseBuilder {
+
+        public SkillResponseBuilder() {
+            skillPayload.addProperty("version", "2.0");
+            skillPayload.add("template", template);
+            template.add("outputs", outputs);
+            template.add("quickReplies", quickReplies);
+        }
+
+        private final JsonArray outputs = new JsonArray();
+        private final JsonObject template = new JsonObject();
+        private final JsonArray quickReplies = new JsonArray();
+        private final JsonObject skillPayload = new JsonObject();
+
+        public SkillResponseBuilder simpleText(String text) {
+
+            if (outputs.size() >= OUTPUTS_LIMIT) {
+                throw new KakaoSkillFailedException("outputs의 제한은 1개 이상 3개 이하입니다.");
+            }
+            JsonObject field = new JsonObject();
+            JsonObject type = new JsonObject();
+            field.addProperty("text", text);
+            type.add(SIMPLE_TEXT, field);
+            outputs.add(type);
+            return this;
+        }
+
+        public SkillResponseBuilder quickReply(String label, String action, String messageText) {
+            if (quickReplies.size() >= QUICK_REPLIES_LIMIT) {
```

Comment:
별건 아니지만 if로직이 길어지니 메서드를 빼서 쓰는거도좋을거 같아요
```suggestion
            public void validateQuickReplies(List<QuickReply> quickReplies, String action, String label) {
    if (quickReplies.size() >= QUICK_REPLIES_LIMIT) {
        throw new KakaoSkillFailedException("quickReplies의 제한은 10개 이하입니다.");
    }
    if (!QUICK_ACTION_MESSAGE.equals(action)) {
        throw new KakaoSkillFailedException("quickReplies의 action이 올바르게 설정되지 않았습니다.");
    }
    if (label.length() > MAX_QUICK_REPLIES_LABEL_SIZE) {
        throw new KakaoSkillFailedException("quickReplies에서 label은 최대 8자 제한입니다.");
    }
}
```

Reply:
Response는 굳이인가..?(이런건 service에서 처리하는것도 좋알을거같다는 생각..?)

Reply:
builder로 구성되어있는 로직에서 검증이라서 private 메소드로 분리하면 되려 가독성에서 이점이 사라진다고 생각했습니다.
아래갔다 위에봤다..